parts/windowsProvision.ps1

```diff
@@ -50,115 +298,137 @@ function InstallOpenSSH()
+    $installDir = Join-Path $env:ProgramFiles "Git"
+    Install-CITool -InstallerPath $PACKAGES["git"]["local_file"] `
+                   -InstallDirectory $installDir `
+                   -ArgumentList @("/SILENT") `
+                   -EnvironmentPath @("$installDir\cmd", "$installDir\bin", "installDir\mingw64\bin")
+
+}
+
+function Install-7Zip {
+    $installDir = Join-Path $env:ProgramFiles "7-Zip"
 
-$git_uri = "https://github.com/git-for-windows/git/releases/download/v2.19.1.windows.1/Git-2.19.1-64-bit.exe"
-$git_file = "c:/tmp/git-2.19.1-64-bit.exe"
-[Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12
-Invoke-WebRequest -Uri $git_uri -Outfile $git_file
-Start-Process -Wait -FilePath $git_file -ArgumentList "/silent /log:c:/tmp/git-install.log"
+    Start-FileDownload -Destination ${seven_zip_file} -URL ${seven_zip_uri}
+    Install-CITool -InstallerPath $PACKAGES["7z"]["local_file"] `
+                   -InstallDirectory $installDir `
+                   -ArgumentList @("/quiet", "/passive") `
+                   -EnvironmentPath @($installDir)
+}
 
-[Environment]::SetEnvironmentVariable("PATH", "$env:PATH;c:/program files/git/mingw64/bin;c:/program files/git/bin;c:/program files/git;c:/program files/7-zip;c:\program files\cmake\bin;C:\Program Files\ocpwin64\4.02.1+ocp1-msvc64-20160113\bin", "Machine")
-[Environment]::SetEnvironmentVariable("PATH", "$env:PATH;c:/program files/git/mingw64/bin;c:/program files/git/bin;c:/program files/git;c:/program files/7-zip", "Process")
-[Environment]::SetEnvironmentVariable("VS150COMNTOOLS", "C:\Program Files (x86)\Microsoft Visual Studio\2017\BuildTools\Common7\Tools", "Machine")
+function Install-SGX {
+    $installDir = Join-Path $PACKAGES_DIRECTORY "sgx_base"
+    Install-ZipCITool -ZipPath $PACKAGES["sgx"]["local_file"] `
+                      -InstallDirectory $installDir
+    & pnputil /add-driver "$installDir\sgx_base.inf"
+    if ($LASTEXITCODE -ne 0) {
+        throw "Failed to install SGX Drivers"
+    }            
+}
 
-##
-#  Install 7zip for unpacking zip and tar files
-#
-$seven_zip_uri = "https://www.7-zip.org/a/7z1805-x64.msi"
-$seven_zip_file = "c:/tmp/7z1805-x64.msi"
-& curl.exe -o $seven_zip_file  $seven_zip_uri 
-Start-Process -Wait -FilePath $seven_zip_file -ArgumentList " /quiet /passive"
+function Install-PSW {
+    $installDir = Join-Path $PACKAGES_DIRECTORY "intel_psw_install"
+    Install-ZipCITool -InstallerPath $PACKAGES["psw"]["local_file"] `
+                      -InstallDirectory $PACKAGES_DIRECTORY
 
-#  Install OpenSSH
-InstallOpenSSH
+    $psw_installer = Join-Path $PACKAGES_DIRECTORY "Intel SGX PSW for Windows v2.1.100.46245\PSW\Intel(R)_SGX_Windows_x64_PSW_2.1.100.46245.exe"
+    Start-Process -Wait -FilePath ${psw_installer} -ArgumentList "--extract-folder $installDir --x"
+    Start-Process -Wait -FilePath "$installDir\setup" -ArgumentList "install --eula=accept --output=$installDir\intel_install.log --components=all"

```

Comment:
Is this a `setup.exe`, `setup.cmd` ? I think it's best if we add the extension as well (if present).